src/providers/definition.ts

```diff
@@ -10,15 +11,15 @@ export class DefinitionProvider implements vscode.DefinitionProvider {
 
 	provideDefinition(document: vscode.TextDocument, position: vscode.Position): vscode.Location | undefined {
 		// look for an @ at the start of the current word (start) and the end of the current word or line (end)
```

Comment:
I can try to help with the `regex`, but first I need to make sure I understand how the matching should happen. Based on lines `14` and `15` I deduce that you need two positions:

- `startResult`: match words that follow after `@` until a space or line end
- `endResult`: what should it match?

One final question, can we use a positive lookbehind assertion? Based on [this page](https://www.regular-expressions.info/javascript.html), it seems to be possible from ECMAScript 2018.

Would the following work (i.e., see [demo here](https://regex101.com/r/q2lwxb/1))?

- `startResult`: `(?<=@)\w+`, or to allow for `-` in the citation key `(?<=@)[\w|-]+`

Reply:
But personally, I find your option already very good (i.e., [demo here](https://regex101.com/r/eeayP8/1)).